pkg/controller/build/build_controller.go

```diff
@@ -156,6 +156,16 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	}
 
+	// validate if "spec.runtime" attributes are valid
+	if b.Spec.Runtime.Base.ImageURL != "" {
```

Comment:
You have this condition a couple of times. My natural comment would be:

NPE if `b.Spec.Runtime` or `b.Spec.Runtime.Base` is `nil`

But sometimes, golang and k8s objects still surprise me. Can't it really be nil? The whole `runtime` section is optional.

Reply:
Initially `b.Spec.Runtime` was not a pointer, but considering this attribute is optional, it should be. I'll make changes accordingly, be careful to inspect for an NPE scenario. Thanks!